onvif/src/soap/client.rs

```diff
@@ -122,6 +129,8 @@ impl Debug for Credentials {
     }
 }
 
+pub type ResponsePatcher = fn(&str) -> String;
```

Comment:
Would it make sense to have this `Box<dyn Fn(&str) -> String>` instead, or is there no useful case where a closure is needed?

Also can this fail sometimes, should it return `Result<String, ...>`?

Reply:
Yeah, probably closures can be useful.

Failures may be useful too. But can't think of different error variants, maybe `Option<String>` is enough.

Reply:
`Result<String, String>` which you did seems good. There could also be `Result<String, Box<dyn Error>>` but that doesn't add any benefit if it is not passed through to the user somehow (aka no benefit when just writing to the logs light it is done right now).